isis/tests/FunctionalTestsIsisImportKaguyaMI.cpp

```diff
@@ -0,0 +1,333 @@
+  EXPECT_DOUBLE_EQ(cube.base(), 0.0);
+  EXPECT_DOUBLE_EQ(cube.multiplier(), 0.013);
+
+  // Instrument Group
+  PvlGroup &inst = isisLabel->findGroup("Instrument", Pvl::Traverse);
+  EXPECT_EQ(inst["MissionName"][0].toStdString(), "SELENE");
+  EXPECT_EQ(inst["SpacecraftName"][0].toStdString(), "KAGUYA");
+  EXPECT_EQ(inst["InstrumentName"][0].toStdString(), "Multiband Imager Near Infrared");
+  EXPECT_EQ(inst["InstrumentId"][0].toStdString(), "MI-NIR");
+  EXPECT_EQ(inst["TargetName"][0].toStdString(), "MOON");
+  EXPECT_EQ(inst["StartTime"][0].toStdString(), "2008-09-16T20:10:30.480257");
+  EXPECT_EQ(inst["StopTime"][0].toStdString(), "2008-09-16T20:10:42.921232");
+  EXPECT_EQ(inst["SpacecraftClockStartCount"][0].toStdString(), "905631021.132");
+  EXPECT_EQ(inst["SpacecraftClockStopCount"][0].toStdString(), "905631033.574");
+  EXPECT_DOUBLE_EQ(inst["LineExposureDuration"], 13.2148);
+  EXPECT_EQ(inst["LineExposureDuration"].unit(), "msec");
+  EXPECT_DOUBLE_EQ(inst["LineSamplingInterval"], 39);
+  EXPECT_EQ(inst["LineSamplingInterval"].unit(), "msec");
+  EXPECT_DOUBLE_EQ(inst["CorrectedSamplingInterval"], 38.999922);
+  EXPECT_EQ(inst["CorrectedSamplingInterval"].unit(), "msec");
+
+  // Archive Group
+  PvlGroup &arch = isisLabel->findGroup("Archive", Pvl::Traverse);
+
+  EXPECT_EQ(arch["DataSetId"][0].toStdString(), "MI-NIR_Level2B");
+  EXPECT_EQ(arch["ProductSetId"][0].toStdString(), "MI-NIR_Level2B2");
+
+  // Bandbin Group
+  PvlGroup &bandBin = isisLabel->findGroup("BandBin", Pvl::Traverse);
+  std::istringstream bandBinStream(R"(
+  Group = BandBin
+    FilterName = (MN1, MN2, MN3, MN4)
+    Center     = (1000.0, 1049.0, 1248.0, 1548.0) <nm>
+    Width      = (27.0, 28.0, 33.0, 48.0) <nm>
+    BaseBand   = MN1
+  End_Group
+  )");
+  PvlGroup bandBinTruth;
+  bandBinStream >> bandBinTruth;
+  EXPECT_PRED_FORMAT2(AssertPvlGroupEqual, bandBin, bandBinTruth);
+
+  // Kernels Group
+  PvlGroup &kern = isisLabel->findGroup("Kernels", Pvl::Traverse);
+  EXPECT_EQ(int(kern["NaifCkCode"]), -131340);
+  EXPECT_EQ(int(kern["NaifFrameCode"]), -131341);
+
+  std::unique_ptr<Histogram> hist (cube.histogram());
+
+  EXPECT_DOUBLE_EQ(hist->Average(), 29.395262812500022);
+  EXPECT_DOUBLE_EQ(hist->Sum(), 188129.68200000015);
+  EXPECT_EQ(hist->ValidPixels(), 6400);
+  EXPECT_DOUBLE_EQ(hist->StandardDeviation(), 2.8449125231835715);
+}
+
+TEST_F(TempTestingFiles, FunctionalTestIsisImportKaguyamiProj) {
+  QTemporaryDir prefix;
+  QString cubeFileName = prefix.path() + "/kaguyatc2isisTEMP.cub";
+  QVector<QString> args = {"from=data/kaguyami2isis/3C5_label.pvl", "to="+cubeFileName};
+
+  UserInterface options(APP_XML, args);
+  try {
+    isisimport(options);
+    FAIL() << "Should not have been able to ingest: " << args[0].toStdString().c_str() << std::endl;
+  }
+  catch (IException &e) {
+    EXPECT_THAT(e.what(), HasSubstr("**PROGRAMMER ERROR** Unable to create a cube label from"));
+  }
+}
+
+/*
```

Comment:
I think we should put in a comment saying why these commented out sections exist and indicate what we are waiting for